src/lib/templateResource.ts

```diff
@@ -0,0 +1,40 @@
+type ResourceReducer =
+  | ((resource: string, config: Config) => string)
+  | ((resource: string) => string);
+
+// Removes all spaces from string, this allows the other reducers to have more
+// reliable match patterns as they don't need to be concerned about whitespace.
+function removeWhitespace(resource: string): string {
+  return resource.replace(/\s/g, "");
+}
+
+// Replaces the template placeholder <% RANDOM %> with a random string
+function replaceRandom(resource: string): string {
+  const dictionary =
```

Comment:
Perf nit: can pull `dictionary` out into constant so it is not regenerated constantly